pkg/storage/kubernetes/ipam.go

```diff
@@ -185,46 +159,142 @@ func (i *KubernetesIPAM) Close() error {
+	for _, r := range reservelist {
+		index, err := iphelpers.IPGetOffset(r.IP, firstip)
+		if err != nil {
+			return nil, err
+		}
+		allocations[fmt.Sprintf("%d", index)] = whereaboutsv1alpha1.IPAllocation{ContainerID: r.ContainerID, PodRef: r.PodRef, Interface: r.IfName}
+	}
+	return allocations, nil
+}
+
 // KubernetesOverlappingRangeStore represents a OverlappingRangeStore interface
 type KubernetesOverlappingRangeStore struct {
-	client      wbclient.Interface
-	containerID string
-	namespace   string
+	client    wbclient.Interface
+	namespace string
 }
 
 // GetOverlappingRangeStore returns a clusterstore interface
 func (i *KubernetesIPAM) GetOverlappingRangeStore() (storage.OverlappingRangeStore, error) {
-	return &KubernetesOverlappingRangeStore{i.client, i.containerID, i.namespace}, nil
+	return &KubernetesOverlappingRangeStore{i.client, i.namespace}, nil
 }
 
 // IsAllocatedInOverlappingRange checks for IP addresses to see if they're allocated cluster wide, for overlapping
-// ranges.
-func (c *KubernetesOverlappingRangeStore) IsAllocatedInOverlappingRange(ctx context.Context, ip net.IP,
-	networkName string) (bool, error) {
-	normalizedIP := normalizeIP(ip, networkName)
+// ranges. First return value is true if the IP is allocated, second return value is true if the IP is allocated to the
+// current podRef
+func (c *KubernetesOverlappingRangeStore) GetOverlappingRangeIPReservation(ctx context.Context, ip net.IP,
+	podRef, networkName string) (*whereaboutsv1alpha1.OverlappingRangeIPReservation, error) {
+	normalizedIP := NormalizeIP(ip, networkName)
 
-	logging.Debugf("OverlappingRangewide allocation check; normalized IP: %q, IP: %q, networkName: %q",
+	logging.Debugf("Get overlappingRangewide allocation; normalized IP: %q, IP: %q, networkName: %q",
 		normalizedIP, ip, networkName)
 
-	_, err := c.client.WhereaboutsV1alpha1().OverlappingRangeIPReservations(c.namespace).Get(ctx, normalizedIP, metav1.GetOptions{})
+	r, err := c.client.WhereaboutsV1alpha1().OverlappingRangeIPReservations(c.namespace).Get(ctx, normalizedIP, metav1.GetOptions{})
 	if err != nil && errors.IsNotFound(err) {
 		// cluster ip reservation does not exist, this appears to be good news.
-		// logging.Debugf("IP %v is not reserved cluster wide, allowing.", ip)
-		return false, nil
+		return nil, nil
 	} else if err != nil {
 		logging.Errorf("k8s get OverlappingRangeIPReservation error: %s", err)
-		return false, fmt.Errorf("k8s get OverlappingRangeIPReservation error: %s", err)
+		return nil, fmt.Errorf("k8s get OverlappingRangeIPReservation error: %s", err)
 	}
 
 	logging.Debugf("Normalized IP is reserved; normalized IP: %q, IP: %q, networkName: %q",
 		normalizedIP, ip, networkName)
-	return true, nil
+	return r, nil
 }
 
 // UpdateOverlappingRangeAllocation updates clusterwide allocation for overlapping ranges.
 func (c *KubernetesOverlappingRangeStore) UpdateOverlappingRangeAllocation(ctx context.Context, mode int, ip net.IP,
```

Comment:
Are you deprecating the containerID field of the OverlappingRangeAllocation?

Reply:
I'm not using it anymore. It was useless. The code only care about podRef. This is good because we do not have to update it. I was going to leave only podRef for these cluster wide reservations but I decided to add ifname. When the pod has multiple interfaces, it is nice to see which allocation belongs to which interface. Just for presentation (and to avoid customers asking why there allocations with duplicated pod references), it does not add any functionality.